inc/class-wpseo-replace-vars.php

```diff
@@ -1395,4 +1396,48 @@ public function get_terms( $id, $taxonomy, $return_single = false ) {
 		 */
 		return apply_filters( 'wpseo_terms', $output, $taxonomy );
 	}
+
+	/**
+	 * Gets a taxonomy term hierarchy including the term to get the parents for.
+	 *
+	 * @return string
+	 */
+	private function get_term_hierarchy() {
+		if ( ! is_taxonomy_hierarchical( $this->args->taxonomy ) ) {
+			return '';
+		}
+
+		$separator = ' ' . $this->retrieve_sep() . ' ';
+
+		$args = [
+			'format'    => 'name',
+			'separator' => $separator,
+			'link'      => false,
+			'inclusive' => true,
+		];
+
+		return rtrim(
+			get_term_parents_list( $this->args->term_id, $this->args->taxonomy, $args ),
+			$separator
+		);
+	}
+
+	/**
+	 * Retrieves the term ancestors hierarchy.
+	 *
+	 * @return string|null The term ancestors hierarchy.
+	 */
+	private function retrieve_term_hierarchy() {
+		$replacement = null;
+
+		if ( ! isset( $replacement ) && isset( $this->args->term_id ) && ! empty( $this->args->taxonomy ) ) {
```

Comment:
A bit confusing why there is an (unnecessary) `! isset( $replacement )` check in the if-statement, when just before `$replacement` is set to `null`. 

Some other methods in this class seem to this as well (e.g., `retrieve_parent_title()`, `retrieve_searchphrase()`), but not all methods. @afercia is looking into it.

Reply:
Looks like some `static $replacement;` were changed to `$replacement = null;` in https://github.com/Yoast/wordpress-seo/commit/5af39a1ec1bc26505be6c8405e088452607f9137 but then the following code still checks for `! isset( $replacement )` which seems redundant. On the other hand, other methods still use `static $replacement;` and I'm not sure to understand why :) Would appreciate some feedback from seniors.

Reply:
I've asked for input from the architects, but this will be addressed in a separate issue.